tox.ini

```diff
@@ -47,13 +43,17 @@ commands =
     make linkcheck
 
 [testenv:sty]
+setenv =
+    SKIP = mypy
 whitelist_externals =
+    mypy
     pre-commit
 commands =
+    mypy .  # run separately because of potential caching problems
```

Comment:
which caching problems?

Reply:
Sometimes `mypy` messes up if you run it with `pre-commit run mypy -a` (I think because that would run over all files under version control).

Reply:
so what is different to calling `mypy .` then?

Reply:
`mypy .` just runs over the files it can find according to these rules
https://github.com/ComPWA/tensorwaves/blob/6ccb5c8ea4f29509c8e330742a381d7c92a230f7/setup.cfg#L90-L97

The `pre-commit` hook also has some sort of filter like that
https://github.com/ComPWA/tensorwaves/blob/6ccb5c8ea4f29509c8e330742a381d7c92a230f7/.pre-commit-config.yaml#L53-L58

but somehow it doesn't work well. This solution worked well so far in the `expertsystem`.

Reply:
so in `pre-commit` those can be used but dont work in the same way?

Reply:
`pre-commit` for `mypy` works if it's just running over staged files. Just _can_ run into problems when running over all files (`run -a`).

Reply:
As a side note: I have the impression that `pre-commit run mypy -a` fails when the interfaces change. Then, if you run that command again, the problem disappears. In contrast, `mypy .` never causes problems.

Reply:
Hmm ok.